.github/workflows/windows.yml

```diff
@@ -468,7 +460,7 @@ jobs:
 
       - name: Upload Test Results
         uses: actions/upload-artifact@v3
-        if: ${{ always() }}
+        if: ${{ always() && !cancelled()}}
```

Comment:
```suggestion
        if: ${{ !cancelled()}}
```

Reply:
then this step will be skipped if one of the previous steps have been failed. 
We need to collect logs if steps are failed

Reply:
Wouldn't it be useful to collect some artifacts, if available, from the cancelled jobs? E.g. if they fail with timeout